devito/passes/iet/mpi.py

```diff
@@ -368,3 +339,55 @@ def mpiize(graph, **kwargs):
         make_halo_exchanges(graph, mpimode=mpimode, **kwargs)
 
     make_reductions(graph, mpimode=mpimode, **kwargs)
+
+
+# *** Utilities
+
+def _filter_iter_mapper(iet):
+    """
+    Given an IET, return a mapper from Iterations to the HaloSpots.
+    Additionally, filter out Iterations that are not of interest.
+    """
+    iter_mapper = MapNodes(Iteration, HaloSpot, 'immediate').visit(iet)
+    iter_mapper = {k: [hs for hs in v if not hs.halo_scheme.is_void]
```

Comment:
nitpicking: three consecutive comprehensions, you may be better off with an explicit for loop (aside from the performance considerations since these lists are tiny)